pkg/schema/fake_registry.go

```diff
@@ -12,9 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build !integration
```

Comment:
The fake schema registry was previously meant to only run in unit tests, now we include it in Conduit because we use it as the in-memory implementation. This all makes sense. However, previously _integration_ tests were running against a real schema registry, meaning that the same test was ensuring that it works in a real-world use case, while allowing the developer to run fast unit tests without spinning up a schema registry. Notice that [`integration.go`](https://github.com/ConduitIO/conduit/blob/main/pkg/plugin/processor/builtin/impl/avro/schemaregistry/integration.go) and [`fake.go`](https://github.com/ConduitIO/conduit/blob/main/pkg/plugin/processor/builtin/impl/avro/schemaregistry/fake.go) both define the functions `ExampleSchemaRegistryURL` and `TestSchemaRegistryURL`, so depending on the flag `--tags=integration` the test either targets the fake or a real schema registry. This behavior should be retained.

Essentially we need to extract everything between lines 31 and 88 into a separate file that's guarded with the `go:build` tag I'm commenting on, and then have another file like [`integration.go`](https://github.com/ConduitIO/conduit/blob/main/pkg/plugin/processor/builtin/impl/avro/schemaregistry/integration.go) that just returns the URL to the real schema registry.

Reply:
Things are easy to miss with the build tags.:)

Reply:
Done!